cmd/catchupsrv/download.go

```diff
@@ -44,34 +48,83 @@ var connsFlag = flag.Int("conns", 2, "Number of connections per server")
 var serverList []string
 var nextBlk uint64
 
-func blockToString(blk uint64) string {
-	return strconv.FormatUint(blk, 36)
+// padLeftZeros pad the string s with zeros on the left to the length n
+func padLeftZeros(s string, n int) string {
+	return strings.Repeat("0", n-len(s)) + s
+}
+
+// blockToFileName converts a block number into the filename in which it will be downloaded
+// When using subfolders, the string corresponding to the base-36 number padded with zeros
+// Otherwise it is just the base-36 number
+func blockToFileName(blk uint64) string {
+	if *subfoldersFlag {
+		return padLeftZeros(strconv.FormatUint(blk, 36), lenBlockStr)
+	} else {
+		return strconv.FormatUint(blk, 36)
+	}
+}
+
+// stringToBlock converts a base-36 string into a block number
+func stringToBlock(s string) uint64 {
+	blk, err := strconv.ParseUint(s, 36, 64)
+	if err != nil {
+		panic(err)
```

Comment:
This panic is reachable from the HTTP request handler, which is undesirable.

Reply:
Thanks! Normally fixed.